### ISSUE TITLE
SMN, Carbuncle, Healing Ruby, incorrect target.  (Only heals self)

### ISSUE THREAD

**Issue by waterlgndx** [DarkstarProject/darkstar#403]
The target for Healing Ruby should be any party member and should not just be limited to self.  ( http://wiki.ffxiclopedia.org/wiki/Healing_Ruby ).  

While testing this I added debugging lines to healing_ruby.lua.  OnAbilityCheck target seems to correctly report the target however OnPetAbility target reports the incorrect target.  I'm not really sure where to go from here but I just figured I'd report this information.

**Comment by waterlgndx** [comment]
I decided that I would get visual studio and try to debug this issue further.  (Despite summoner's incomplete status, it's my baby.)
I've been looking at ai_pet_dummy.cpp and I've been testing to see where the issue might be.  So far I've found this:
for healing_ruby CAIPetDummy::preparePetAbility
preconditions:
- PTarg starts as Carbuncle
- m_PBattleSubTarget starts as the proper party member target.

postconditions:
- m_PBattleSubTarget becomes the pet's master.

Ln 311, ai_pet_dummy
```
else if (m_PMobSkill->getValidTargets() & TARGET_PLAYER_PARTY)
        {
			// Only overwrite the sub target if it it not specified or
			// the input target doesn't match the sub target.
			if (m_PBattleSubTarget == nullptr || PTarg != m_PBattleSubTarget) {
				m_PBattleSubTarget = m_PPet->PMaster;
			}
        }
```

so I'm thinking this is always going to be true for player party targets.  This isn't a huge deal since most blood pacts aren't player targeted (`TARGET_PLAYER_PARTY`) and it doesn't have an impact on avatar positioning either (I'm pretty sure it'll still go towards the target you specify at first).  

I guess I'm just not sure if PTarg should be Carbuncle at this point (or whatever pet) or if it should be something else.  So if this isn't the desired behavior then it needs changed somewhere.  I'm also not sure if it's something simple like the author originally wanted it to be an and or had some other ideas for this.  I haven't contributed code yet anywhere so I'm not sure if I'm comfortable trying to fix it.  But I may be adjusting this for myself at least.
